File: Android/shared/src/main/kotlin/shared/dto/objects/Statistic.kt
@file:UseSerializers(
    UUIDSerializer::class,
    DateISOSerializer::class
)

package shared.dto.objects

import shared.serializers.DateISOSerializer
import kotlinx.serialization.Serializable
import kotlinx.serialization.UseSerializers
import shared.dto.primitives.Price
import shared.dto.protocols.Parametable
import shared.dto.protocols.Responsable
import shared.serializers.UUIDSerializer
import java.util.*

object Statistic {
    data object Parameters {
        @Serializable
        data class AppointmentsQuery(
            val startDate: Date,
            val endDate: Date,
            val employees: List<UUID>? = null,
            val salons: List<UUID>? = null,
        ) : Parametable()
    }

    data object Responses {
        @Serializable
        data class Appointments(
            val price: Price,
            val count: Int,
        ) : Responsable
    }
}

File: IOS/DTOs/Objects/Statistic.swift
import Foundation
import MemberwiseInit

public enum Statistic {
    public enum Parameters {}
    public enum Responses {}
}

public extension Statistic.Parameters {
    
    @MemberwiseInit(.public, _optionalsDefaultNil: true)
    struct AppointmentsQuery: Parametable {
        public let startDate: Date
        public let endDate: Date
        public let employees: [UUID]?
        public let salons: [UUID]?
    }
}

public extension Statistic.Responses {
    
    @MemberwiseInit(.public, _optionalsDefaultNil: true)
    struct Appointments: Responsable {
        public var price: Price
        public var count: Int
    }
}
